ngraph/core/src/partial_shape.cpp

```diff
@@ -23,26 +23,53 @@
 
 using namespace ngraph;
 
+PartialShape::PartialShape()
+    : PartialShape(std::initializer_list<Dimension>{})
+{
+}
+
+PartialShape::PartialShape(std::initializer_list<Dimension> init)
+    : PartialShape(true, init)
+{
+}
+
 PartialShape::PartialShape(const std::vector<Dimension::value_type>& dimensions)
     : m_rank_is_static(true)
+    , m_dimensions(dimensions.begin(), dimensions.end())
 {
-    std::transform(dimensions.cbegin(),
-                   dimensions.cend(),
-                   std::back_inserter(m_dimensions),
-                   [](const Dimension::value_type& dimension) { return dimension; });
 }
 
 PartialShape::PartialShape(const Shape& shape)
-    : PartialShape(true, {})
+    : m_rank_is_static(true)
+    , m_shape_type(ShapeType::SHAPE_IS_STATIC)
+    , m_dimensions(shape.begin(), shape.end())
+{
+}
+
+PartialShape::PartialShape(bool rank_is_static, const std::vector<Dimension>& dimensions)
+    : m_rank_is_static(rank_is_static)
+    , m_dimensions(dimensions)
```

Comment:
For this case `m_shape_type` is unknown? Can we check the vector of dimensions in order to understand the status?

Reply:
Yes, we can but I decided to use lazy evaluation of shape type. Perhaps, in some cases, the is_static method is not needed.